# Dataset Understanding 📊 

This dataset represents user clickstream behavior in an e-commerce clothing store (2008).

Each row corresponds to a user interaction within a session, including:

- Product viewed
- Page location
- Navigation order
- Product attributes (price, color, category)

**The dataset enables analysis of:**

- User navigation patterns
- Product preferences
- Impact of UI placement on behavior
- Differences across countries

In [25]:
#Imports
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [26]:
import pandas as pd

df = pd.read_csv("../data/e_shop_clothing_2008.csv", sep =";")
df.head()

,year,month,day,order,country,session ID,page 1 (main category),page 2 (clothing model),colour,location,model photography,price,price 2,page
0,2008,4,1,1,29,1,1,A13,1,5,1,28,2,1
1,2008,4,1,2,29,1,1,A16,1,6,1,33,2,1
2,2008,4,1,3,29,1,2,B4,10,2,1,52,1,1
3,2008,4,1,4,29,1,2,B17,6,6,2,38,2,1
4,2008,4,1,5,29,1,2,B8,4,3,2,52,1,1


In [27]:
# Check dataset structure: data types, null values, and memory usage
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 165474 entries, 0 to 165473
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype
---  ------                   --------------   -----
 0   year                     165474 non-null  int64
 1   month                    165474 non-null  int64
 2   day                      165474 non-null  int64
 3   order                    165474 non-null  int64
 4   country                  165474 non-null  int64
 5   session ID               165474 non-null  int64
 6   page 1 (main category)   165474 non-null  int64
 7   page 2 (clothing model)  165474 non-null  str  
 8   colour                   165474 non-null  int64
 9   location                 165474 non-null  int64
 10  model photography        165474 non-null  int64
 11  price                    165474 non-null  int64
 12  price 2                  165474 non-null  int64
 13  page                     165474 non-null  int64
dtypes: int64(13), str(1)
memory usage: 17.7 MB


In [28]:
# Analyze number of unique sessions to estimate total user interactions
df["session ID"].nunique()

24026

In [29]:
# Compute session depth (number of clicks per session)
# This helps identify highly engaged users
df.groupby("session ID")["order"].max().describe()

count    24026.000000
mean         6.887289
std          8.995161
min          1.000000
25%          2.000000
50%          4.000000
75%          8.000000
max        195.000000
Name: order, dtype: float64

In [30]:
# Analyze distribution of product categories to understand user interest
df["page 1 (main category)"].value_counts()

page 1 (main category)
1    49742
4    38747
3    38577
2    38408
Name: count, dtype: int64

In [31]:
# Evaluate how product placement (LOCATION) affects user interaction
df["location"].value_counts()

location
1    34532
2    33383
5    27783
4    27377
3    21656
6    20743
Name: count, dtype: int64

### Insight: Dataset Quality

The dataset contains 165,474 user interactions across 24,026 sessions.

There are no missing values, indicating a clean and reliable dataset for analysis

Most variables are numerical, with only one categorical feature (clothing model)

### Insight: User Engagment

Most user sessions are relatively short, with a median of 4 interactions per session.

However, there are extreme cases with up to 195 clicks, indicating the presence of highly engaged users.

This suggests a skewed distribution where a small group of users contributes significantly to overall interaction


### Insight: Product Category Preference

Trousers (category 1) receive the highest number of interactions, making them the most popular product category

Sales item (category 4) also show strong engagement, suggesting that discounts significantly influence user behavior

Other categories (skirts and blouses) have similar interaction levels, indicating balanced interest

### Insight: Impact of Product Placement

Products displayed in the top-left and top-middle sections receive the highest number of interactions

In contrast, items located in the bottom-right area receive the least attention

This indicates that visibility and position on the page strongly influece user engagement, highlighting the importance of UI design in e-commerce plarforms